core/src/main/scala/com/spingo/op_rabbit/ConnectionParams.scala

```diff
@@ -84,4 +84,60 @@ object ConnectionParams {
       ssl = config.getBoolean("ssl")
     )
   }
+
+  private def fromUri(uri: URI): ConnectionParams = {
+    val uriAuthority = uri.getAuthority
+    val (hosts, (username, password)) = uriAuthority.indexOf('@') match {
+      case -1 =>
+        parseUriHosts(uriAuthority) -> parseUserPassword(None)
+      case idx =>
+        parseUriHosts(uriAuthority.substring(idx + 1)) -> parseUserPassword(Some(uriAuthority.substring(0, idx)))
+    }
+    val params = Option(uri.getQuery).fold(Map.empty[String, String]) { query =>
```

Comment:
I'm heavily in favor of "fail loud and proud". If the user specifies a parameter that is not support, we should fail. Can you make that a reality?

Reply:
Absolutely, I was thinking to not be too strict for our users, but your proposal indeed makes sense. Follow the updates :)